docker/cli
# Fix docker plugin suggestions in Zsh completion

Fixes #2761

**- What I did**

The `__docker_plugins` function calls `docker plugin ls` to get its information, but it didn't parse it correctly before, so I fixed it. Without this change, triggering the function would produce the following errors:

```
__docker_plugins:27: bad math expression: empty string
__docker_plugins:27: bad math expression: empty string
__docker_plugins:27: bad math expression: empty string
__docker_plugins:27: bad math expression: empty string
__docker_plugins:27: bad math expression: empty string
```

**- How I did it**

The `docker plugin ls` command doesn't output a `TAG` column; instead, its name is `ID`. So one of the changes is to replace `TAG` for `ID`. Also, the relevant part of ID hash is the left one, so we need to use right-padding to trim it to 7 characters. With `(l:7:: :::)` what we get is the last 7 characters of the hash, which is not what we want.

**- How to verify it**

1. If you don't have docker plugins installed, install at least 2. For example:
   ```sh
   docker plugin install vieux/sshfs
   docker plugin install grafana/loki-docker-driver:latest --alias loki --grant-all-permissions
   ```
2. Run `docker plugin ls` to see the output:
   ```console
   $ docker plugin ls
   ID             NAME                 DESCRIPTION               ENABLED
   a237ec8092c3   loki:latest          Loki Logging Driver       true
   84dd2ad7d54e   vieux/sshfs:latest   sshFS plugin for Docker   true
   ```
3. Make sure to be using this `_docker` file for completion. In Zsh, download this `_docker` file and run this to load it:
   ```zsh
   unfunction _docker
   autoload -Uz /path/to/downloaded/_docker/file
   ```
4. Tab-complete `docker plugin inspect`:
   ```console
   $ docker plugin inspect [TAB]
   loki         -- latest:a237ec8
   vieux/sshfs  -- latest:84dd2ad
   ```

**- Description for the changelog**

Fix `docker plugin` Zsh completion by correctly parsing the output of `docker plugin ls`.

**- A picture of a cute animal (not mandatory but encouraged)**

![cute sleeping baby donkeys](https://user-images.githubusercontent.com/1441704/103438635-fd808200-4c34-11eb-9335-b1fae399e570.png)
